core/src/main/scala/schema/provider/metadata/SchemaConversions.scala

```diff
@@ -91,110 +98,129 @@ trait SchemaConversions {
             Some(Index.Using(metadata.getIndexClassName, None))
           else None
       )
-    } leftMap (SchemaDefinitionProviderError(_))
+    }
+
+  def toUserType[M[_]](userType: UserType)(implicit M: MonadError[M, Throwable]): M[CreateType] = {
 
-  def toUserType(userType: UserType): SchemaResult[CreateType] =
-    Either.catchNonFatal {
-      userType.getFieldNames.asScala.toList.traverse { fieldName =>
-        toField(fieldName, userType.getFieldType(fieldName))
-      } map { list =>
+    val fieldsM: M[List[Field]] =
+      M.flatMap(M.catchNonFatal(userType.getFieldNames.asScala.toList)) { list =>
+        list.traverse { fieldName =>
+          toField(fieldName, userType.getFieldType(fieldName))
+        }
+      }
+
+    M.flatMap(fieldsM) { list =>
+      M.catchNonFatal {
         CreateType(
           ifNotExists = false,
           typeName = TypeName(Some(KeyspaceName(userType.getKeyspace)), userType.getTypeName),
           fields = list)
       }
-    } leftMap (SchemaDefinitionProviderError(_)) joinRight
+    }
```

Comment:
I think this can be something like

```scala
val fieldsM = ...
val typeNameM = M.catchNonFatal(
  TypeName(Some(KeyspaceName(userType.getKeyspace)), userType.getTypeName))
M.map2(fieldsM, typeNameM)(
  (list, tpeName) => CreateType(ifNotExists = false, typeName = tpeName, fields = list))
```